crates/apub/src/protocol/objects/page.rs

```diff
@@ -124,6 +125,26 @@ impl Attachment {
       _ => None,
     }
   }
+
+  pub(crate) async fn as_markdown(&self, context: &Data<LemmyContext>) -> LemmyResult<String> {
+    let (url, name, media_type) = match self {
+      Attachment::Image(i) => (i.url.clone(), i.name.clone(), Some(String::from("image"))),
+      Attachment::Document(d) => (d.url.clone(), d.name.clone(), d.media_type.clone()),
+      Attachment::Link(l) => (l.href.clone(), None, l.media_type.clone()),
+    };
+
+    let is_image =
+      media_type.is_some_and(|media| media.starts_with("video") || media.starts_with("image"));
+
+    if is_image {
+      let url = proxy_image_link_opt_apub(Some(url), context)
+        .await?
+        .expect("No url");
```

Comment:
Better use `proxy_image_link()` which doesnt take option (you need to make it public).

Reply:
It looks like we want that to be crate specific, and have the `_api` and `_apub` versions public.

So really you just need to add a `proxy_image_link_apub` function to `crates/api_common/src/utils.rs`

Reply:
That would just be another wrapper, and there is already `proxy_image_link_internal()`. No need to have another wrapper which does the exact same thing.

Reply:
Alright. I see a bunch of duplicated functions there anyway that could be cleaned up then.

Reply:
> It looks like we want the proxy_image_link function to be crate specific, and have the _api and _apub versions public.

This was my impression as well.